crates/bitwarden/src/vault/cipher/cipher.rs

```diff
@@ -366,23 +368,48 @@ impl CipherView {
         }
     }
 
+    fn reencrypt_attachment_keys(
+        &mut self,
+        old_key: &SymmetricCryptoKey,
+        new_key: &SymmetricCryptoKey,
+        ignore_null_keys: bool,
+    ) -> Result<()> {
+        if let Some(attachments) = &mut self.attachments {
+            // If any attachment is missing a key we can't reencrypt the attachment keys
+            if !ignore_null_keys && attachments.iter().any(|a| a.key.is_none()) {
+                return Err("This cipher contains attachments without keys. Those attachments will need to be reuploaded to complete the operation".into());
+            }
+
+            for attachment in attachments {
+                if let Some(attachment_key) = &mut attachment.key {
+                    let dec_attachment_key: Vec<u8> = attachment_key.decrypt_with_key(old_key)?;
+                    *attachment_key = dec_attachment_key.encrypt_with_key(new_key)?;
+                }
+            }
+        }
+        Ok(())
+    }
+
     pub fn move_to_organization(
         &mut self,
         enc: &dyn KeyContainer,
         organization_id: Uuid,
     ) -> Result<()> {
-        // If the cipher has a key, we need to re-encrypt it with the new organization key
-        if let Some(cipher_key) = &mut self.key {
-            let old_key = enc
-                .get_key(&self.organization_id)
-                .ok_or(Error::VaultLocked)?;
+        let old_key = enc
+            .get_key(&self.organization_id)
+            .ok_or(Error::VaultLocked)?;
 
-            let new_key = enc
-                .get_key(&Some(organization_id))
-                .ok_or(Error::VaultLocked)?;
+        let new_key = enc
+            .get_key(&Some(organization_id))
+            .ok_or(Error::VaultLocked)?;
 
+        // If the cipher has a key, we need to re-encrypt it with the new organization key
+        if let Some(cipher_key) = &mut self.key {
```

Comment:
I think we need to check the attachments here too?

Reply:
Ah yes, I lifted the condition before this point so it covers both cases